cmake/pcl_find_sse.cmake

```diff
@@ -223,6 +217,15 @@ function(PCL_CHECK_FOR_SSE)
       # platforms.
       list(APPEND SSE_FLAGS "-ffloat-store")
     endif()
+    
+    if(PCL_ENABLE_MARCHNATIVE AND (NOT CMAKE_CROSSCOMPILING))
+      if(HAVE_MARCH)
+          list(APPEND SSE_FLAGS "-march=native")
```

Comment:
march=native is not a SSE flag. Thoughts on AVX enabled, but SSE disabled configuration? Can we prevent the user from shooting themselves in the foot?

IMO, march=native is separate from AVX and SSE configurations (AVX and SSE are kinda-standard opsets, but march native can also use non-standard ops)

Reply:
A cleaner solution might be to unify pcl_find_sse.cmake and pcl_find_avx.cmake, and have a single variable (e.g. `ARCH_FLAGS`) that contains sse, avx, and march flags, instead of `SSE_FLAGS` and `AVX_FLAGS`. But I would say the current state of this PR still works, and in the interest of getting this merged soon I would propose to not do any major changes here.
I think the cmake options `PCL_ENABLE_SSE`, `PCL_ENABLE_AVX`, and `PCL_ENABLE_MARCHNATIVE` give good control over which flags are added.